cypress/e2e/po/pages/global-settings/branding.po.ts

```diff
@@ -42,13 +55,26 @@ export class BrandingPagePo extends RootClusterPage {
     return new ColorInputPo('[data-testid="link-color-input"]');
   }
 
+  uploadButton(label: string) {
+    return cy.getId('file-selector__uploader-button').contains(label).next('input');
+  }
+
+  logoPreview(theme: string) {
+    return cy.getId(`branding-logo-${ theme }-preview`);
+  }
+
+  faviconPreview() {
+    return cy.getId('branding-favicon-preview');
+  }
+
   applyButton() {
     return new AsyncButtonPo('[data-testid="branding-apply-async-button"]', this.self());
   }
 
-  applyAndWait(endpoint: string) {
+  applyAndWait(endpoint: string): Cypress.Chainable {
     cy.intercept('PUT', endpoint).as(endpoint);
     this.applyButton().click();
-    cy.wait(`@${ endpoint }`).its('response.statusCode').should('eq', 200);
+
+    return cy.wait(`@${ endpoint }`);
```

Comment:
this drops the check for 200. would we be able to keep it, if not were there specific scenarios why it should be removed?

Reply:
The `Favicon` test required property data checks (in the spec level) which couldn't be done while the `applyAndWait` fn had the status code check in the po which is why I removed it. I can add the status code checks on the test level where i'm calling `applyAndWait` and not checking status code. that work?

Reply:
Something like this should cover all bases. If ok, later on, we should use the same pattern in other places

```
  applyAndWait(endpoint: string, statusCode = 200): Cypress.Chainable {
    cy.intercept('PUT', endpoint).as(endpoint);
    this.applyButton().click();

    const wait = cy.wait(`@${ endpoint }`)

    return statusCode ? wait.its('response.statusCode').should('eq', wait) : wait;
  }
```

Reply:
thanks for the suggestion! updated in https://github.com/rancher/dashboard/pull/9458/commits/c7361c8b9177a820284ed18c7ab5a3d953d955e9

Reply:
with the switch from the default statusCode of 200 to optional it now needs to be supplied

`applyAndWait(endpoint: string, statusCode = 200)` --> `applyAndWait(endpoint: string, statusCode?: number)`

`applyAndWait('xyz')` --> `applyAndWait('xyz', 200)`

i think we should get this pr in, so no need to update now, but worth considering in the future